Tests/ApolloCacheDependentTests/LoadQueryFromStoreTests.swift

```diff
@@ -319,7 +319,44 @@ class LoadQueryFromStoreTests: XCTestCase {
       }
     }
   }
-  
+
+
+  func testLoadingQueryWithFloats() throws {
+    let starshipLength = 1234.5
+
+    let initialRecords: RecordSet = [
+      "QUERY_ROOT": ["starshipLength(length:1234.5)": Reference(key: "starshipLength(length:1234.5)")],
+      "starshipLength(length:1234.5)": ["__typename": "Starship",
+                                        "name": "Millennium Falcon",
+                                        "length": starshipLength,
+                                        "coordinates": [[0.0,1.0], [2.0,3.0]]]
+    ]
+
+
+    withCache(initialRecords: initialRecords) { (cache) in
+      store = ApolloStore(cache: cache)
+
+      let query = StarshipLengthQuery(length: starshipLength)
+
+      load(query: query) { result in
+        switch result {
+        case .success(let graphQLResult):
+          XCTAssertNil(graphQLResult.errors)
+
+          guard let data = graphQLResult.data else {
+            XCTFail("No data returned with result")
+            return
+          }
```

Comment:
You can use `let data = try XCTUnwrap(graphQLResult.data)` here and not have to deal with writing your own failure mechanism

Reply:
Maybe I'm missing something but I don't think I can use that in the completion block.

Reply:
I haven't had an issue with that in other tests

Reply:
It's not used in completion blocks anywhere else, it seems like this test file uses the completion block with XCTFail pattern for all the tests.

Reply:
no worries i'll fight with this once this is merged